src/pke/extras/ckks-bootstrapping-precision.cpp

```diff
@@ -0,0 +1,143 @@
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//==================================================================================
+
+/*
+
+Use this script to find the correction factor, which gives the best precision for CKKS bootstrapping.
+Specifically, we used this to choose the default correction factor for 64-bit FLEXIBLEAUTOEXT.
+
+*/
+
+#define PROFILE
+
+#include "openfhe.h"
+
+using namespace lbcrypto;
+
+double MeasureBootstrapPrecision(uint32_t numSlots, uint32_t correctionFactor);
+
+double CalculateApproximationError(const std::vector<std::complex<double>>& result,
+                                   const std::vector<std::complex<double>>& expectedResult) {
+    if (result.size() != expectedResult.size())
+        OPENFHE_THROW(config_error, "Cannot compare vectors with different numbers of elements");
+
+    // using the Euclidean norm
+    double avrg = 0;
+    for (size_t i = 0; i < result.size(); ++i) {
+        avrg += std::pow(std::abs(result[i].real() - expectedResult[i].real()), 2);
+    }
+
+    avrg = std::sqrt(avrg) / result.size();  // get the average
+    return std::abs(std::log2(avrg));
+}
+
+int main(int argc, char* argv[]) {
+#if NATIVEINT == 64
+    size_t numIterations           = 1;
+    std::vector<uint32_t> slotsVec = {32768};
+    for (uint32_t numSlots : slotsVec) {
+        for (double i = 8; i < 16; i++) {
+            std::cout << "=======================================================================" << std::endl;
+            std::cout << "Number of slots: " << numSlots << std::endl;
+            std::cout << "Correction Factor: " << i << std::endl;
+            double precision = 0.0;
+            for (size_t j = 0; j < numIterations; j++) {
+                precision += MeasureBootstrapPrecision(numSlots, 0);
```

Comment:
I suggest changing the the correction factor from 0 to i. Otherwise, the output is confusing.